### PR TITLE
metadata mapping reported using registry config

### PR DESCRIPTION
**Pull request by sanxun0325**
**What this PR does**:
local mode, metadata mapping is reported using registration information
**Which issue(s) this PR fixes**:

Fixes https://github.com/apache/dubbo-go/issues/1610

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by LaurenceLiZhixin** [comment]
需要在config pkg下增加 import _ "xxx/metadata-report/nacos" 类似的依赖，保证单测通过

File: metadata/mapping/metadata/service_name_mapping.go [hunk 1]
```diff
@@ -18,6 +18,7 @@
 package metadata
 
 import (
+	"dubbo.apache.org/dubbo-go/v3/metadata/report"
```
Comment (AlexStocks) [inline metadata/mapping/metadata/service_name_mapping.go:21]: kick your ass

File: config/instance/registry_metadata_report.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	"dubbo.apache.org/dubbo-go/v3/metadata/report"
+)
+
+var (
+	regInstances = make(map[string]report.MetadataReport, 8)
+	mux          sync.RWMutex
+)
+
+// GetMetadataReportInstanceByReg obtain metadata report instance through registry config
+func GetMetadataReportInstanceByReg(protocol string) report.MetadataReport {
```
Comment (justxuewei) [inline config/instance/registry_metadata_report.go:36]: Is `GetMetadataReportInstanceByProtocol` better than current name?

File: config/instance/registry_metadata_report_test.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	"dubbo.apache.org/dubbo-go/v3/metadata/report/factory"
+)
+
+func TestGetMetadataReportInstanceByReg(t *testing.T) {
+	t.Run("nil", func(t *testing.T) {
+		reportInstance := GetMetadataReportInstanceByReg("mock")
+		assert.Nil(t, reportInstance)
+	})
+
+	t.Run("nil", func(t *testing.T) {
```
Comment (justxuewei) [inline config/instance/registry_metadata_report_test.go:41]: The test name should be "notnil".

Comment (sanxun0325) [reply config/instance/registry_metadata_report_test.go]: done

File: config/instance/registry_metadata_report.go (hunk 2 above)
Comment (sanxun0325) [reply config/instance/registry_metadata_report.go:36]: GetMetadataReportInstanceByRegistryProtocol is better, but it's too long. 
Can discuss

**Review by zhaoyunxing92 — APPROVED** [review]

**Review by Lvnszn — APPROVED** [review]

File: config/instance/registry_metadata_report.go (hunk 2 above)
Comment (AlexStocks) [reply config/instance/registry_metadata_report.go:36]: too naive.  ’GetMetadataReportInstance(RegistryProtocol string)' is enough.

**Review by Mulavar — APPROVED** [review]

File: config/instance/registry_metadata_report.go (hunk 2 above)
Comment (sanxun0325) [reply config/instance/registry_metadata_report.go:36]: > 

Big Old, there's a duplicate name

Comment (AlexStocks) [reply config/instance/registry_metadata_report.go:36]: Then I think the suggestion from @justxuewei is better.

Comment (justxuewei) [reply config/instance/registry_metadata_report.go:36]: In most cases, "reg" means regular expression, and it rarely stands for "registry". IMO,  both "Protocol" or "RegistryProtocol" sound good to me.

Comment (sanxun0325) [reply config/instance/registry_metadata_report.go]: Change to GetMetadataReportByRegistyProtocol

**Pull request merged**